src/vocabularyChecker/detectWrongSyntax.ts

```diff
@@ -0,0 +1,142 @@
+	const positions = new Map();
+	for (const word of invalidWords) {
+		positions.set(word, [text.indexOf(word), text.indexOf(word) + word.length - 1]);
+	}
+	return positions;
+}
+
+/**
+ * (input:) @param text
+ * This method deletes all UNICODE-characters \u2000-\u206F\u2E00-\u2E7F
+ * as well as '!"#$%&()*+,\-.\/:;<=>?@\[\]^_`{|}~]/. and
+ * replaces them with whitespaces. A word is afterwards (in @function splitIntoWords)
+ * considered as a substring with has at least a whitespace on the right
+ * or on the left.
+ * The exact symbols that can be deleted may have to be overthought,
+ * since brackets, commas and fullstops seem to be necessary to make the user input Prolog-readable.
+ */
+
+export function eliminatePunctuation(text: string) {
+		return text.replace(punctuation, " ").replace(space, " ");
+}
+
+/**
+ * (input:) @param text
+ * This method first deletes and leading or trailing whitespaces off of the
+ * input string. It then calls @function findSpacePositionsInStringArray, to
+ * create an array containing the positions where a new word begins.
+ * Currently this adds a whitespace to every word after the first one.
+ * After doing some corrections to the text, the method @returns @param result,
+ * which is an array containing as entries the single words filtered out
+ * of the text.
+ */
+
+export function splitIntoWords(text: string) {
+		text = text.trim();
+		const spacePositions = findSpacePositionsInStringArray(text);
+		let tempPos = 0;
+		const result: string[] = [];
+				// this loop is problematic. Adds a whitespace to the beginning of each word after the second one.
+		spacePositions.forEach(element => {
+				result.push(text.substring(tempPos, element));
+				tempPos = element;
+		});
+						// the substring after the last whitespace seems to miss, if this substring is not empty. Fix in next line.
+		if (text.substring(spacePositions[spacePositions.length - 1], text.length) !== " ") {
+		result.push(text.substring(spacePositions[spacePositions.length - 1], text.length));
+		}
+		for (let index = 0; index < result.length; index++) {
+				if (result[index].startsWith(" ")) {
+						result[index] = result[index].substring(1, result[index].length); // delete the wrong whitespaces in the beginning
+				}
+		}
+		if (!text.replace(/\s/g, "").length) {
+			return [];
+		}
+		return result;
+}
+
+/**
+ * (input:) @param text
+ * Just a function bundling @function eliminatePunctuation and
+ * @function splitIntoWords together for better readibility
+ * and in case, those two methods need to be called more
+ * often in conjunction.
+ */
+
+function preprocessText(text: string) {
+	return removeDuplicates(splitIntoWords(eliminatePunctuation(text)));
+}
+
+/**
+ * Collects all positions where there is a whitespace in a string.
+ */
+
+function findSpacePositionsInStringArray(text: string) {
+	let index = 0;
+	const spacePositions = [];
+	while ((index = text.indexOf(" ", index + 1)) > 0) {
```

Comment:
there is an `includes` method.

Reply:
includes was not a helpful idea. However, I should test the function, if it works correct for texts like "hello,@;)(/ world". Should only result in one space position. Pipeline may be wrong however and return multiple ones, since ,@;)(/ are all replaced with a Whitespace. In that case `trim` beforehand.